src/config/id.rs

```diff
@@ -0,0 +1,122 @@
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::{
+    cmp::{Ord, Ordering, PartialOrd},
+    fmt,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ComponentId {
+    value: String,
```

Comment:
@jdrouet Sorry for this late comment, but I'm wondering what `value` is?

Reply:
I suspect I inadvertently prompted this by suggesting an `.as_str()` implementation.

I don't think this is currently needed. The idea being, we'd introduce an 'inner' value if/when the `id` field is augmented with a secondary field for scope. At that time, the inner value would represent a pre-allocated name for the full id + scope, and we'd avoid multiple to_string() calls.

In the interim, `.as_str()` should just pull from id.

I'll fire off a quick follow-up PR for this small adjustment.

Reply:
My bad, I prepared this for the next PR adding the scope.